src/persist-client/tests/machine/gc

```diff
@@ -0,0 +1,141 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This test uses a simplifying assumption that a batch is
+# always made up of exactly 1 batch part. This is because
+# batch parts are given random UUID names, meaning we can't
+# easily reference them in our expected test results. We
+# instead refer to a batch part by its batch name
+
+write-batch output=b0 lower=0 upper=1
+k1 0 1
+----
+parts=1 len=1
+
+write-batch output=b1 lower=1 upper=2
+k2 1 -1
+k3 1 1
+----
+parts=1 len=2
+
+write-batch output=b2 lower=2 upper=3
+k3 2 -1
+k4 2 1
+----
+parts=1 len=2
+
+# insert our first batch into state
+compare-and-append input=b0
+----
+ok
+
+# referencing seqnos directly is a bit fragile. as any changes
```

Comment:
I don't think we should do this yet, but if it becomes too much of a maintenance burden, we could solve it by "naming" seqnos. e.g. the DatadrivenState could have some map of <String, SeqNo> and you could perhaps "name current seqno" and then use the names here instead of numbers